R/map.R

```diff
@@ -218,6 +221,27 @@ map_raw <- function(.x, .f, ...) {
   .Call(map_impl, environment(), ".x", ".f", "raw")
 }
 
+
+#' @rdname map
+#' @param .ptype If `NULL`, the default, the output type is the common type
+#'   of the elements of the result. Otherwise, supply a "prototype" giving
+#'   the desired type of output.
+#' @importFrom vctrs vec_c vec_size vec_ptype_common
+#' @export
+map_vec <- function(.x, .f, ..., .ptype = NULL) {
+  out <- map(.x, .f, ...)
+
+  .ptype <- vec_ptype_common(!!!out, .ptype = .ptype)
+  for (i in seq_along(out)) {
```

Comment:
Might be worth adding a `size` argument to `vctrs::list_check_all_vectors()`.

Then this'd be:

```r
out <- map(.x, .f, ...)
list_check_all_vectors(out, size = 1, arg = "")
vec_unchop(out)
```

Reply:
In the meantime if you want to use the `stop_bad_element_vector()` error message then I'd suggest

`sizes <- list_sizes(out); any(sizes != 1L)`